File: Octopus.Client/Csharp/Feeds/CreateFeed.cs
// If using .net Core, be sure to add the NuGet package of System.Security.Permissions

// Reference Octopus.Client
//#r "path\to\Octopus.Client.dll"

using Octopus.Client;
using Octopus.Client.Model;

// Declare working varibles
var octopusURL = "https://youroctourl";
var octopusAPIKey = "API-YOURAPIKEY";

string spaceName = "Default";
string feedName = "nuget.org 3";
string feedURI = "https://api.nuget.org/v3/index.json";
int downloadAttempts = 5;
int downloadRetryBackoffSeconds = 10;
bool useExtendedApi = false;
// optional
string feedUsername = "";
string feedPassword = "";

// Create repository object
var endpoint = new OctopusServerEndpoint(octopusURL, octopusAPIKey);
var repository = new OctopusRepository(endpoint);
var client = new OctopusClient(endpoint);

try
{
    // Get space
    var space = repository.Spaces.FindByName(spaceName);
    var repositoryForSpace = client.ForSpace(space);

    var feedResource = new NuGetFeedResource
    {
        SpaceId = space.Id,
        Name = feedName,
        FeedUri = feedURI,
        DownloadAttempts = downloadAttempts,
        DownloadRetryBackoffSeconds = downloadRetryBackoffSeconds,
        EnhancedMode = useExtendedApi
    };
    if (!string.IsNullOrWhiteSpace(feedUsername))
    {
        feedResource.Username=feedUsername;
    }
    if (!string.IsNullOrWhiteSpace(feedPassword))
    {
        feedResource.Password = feedPassword;
    }
    var feed = repositoryForSpace.Feeds.Create(feedResource);
}
catch (Exception ex)
{
    Console.WriteLine(ex.Message);
    return;
}

File: Octopus.Client/Go/Feeds/create_feed.go
package main

import (
	"github.com/OctopusDeploy/go-octopusdeploy/client"
	"github.com/OctopusDeploy/go-octopusdeploy/model"
)

var (
	// Declare working variables
	octopusURL    string = "https://youroctourl"
	octopusAPIKey string = "API-YOURAPIKEY"

	spaceName                   string = "Default"
	feedName                    string = "nuget.org 3"
	feedURI                     string = "https://api.nuget.org/v3/index.json"
	downloadAttempts            int    = 5
	downloadRetryBackoffSeconds int    = 10
	useExtendedAPI              bool   = false
	// optional
	feedUsername string = ""
	feedPassword string = ""
)

func main() {
	client, err := client.NewClient(nil, octopusURL, octopusAPIKey, spaceName)

	if err != nil {
		// TODO: handle error
	}

	space, err := client.Spaces.GetByName(spaceName)

	if err != nil {
		// TODO: handle error
	}

	feedResource := model.NewNuGetFeed(feedName)

	if err != nil {
		// TODO: handle error
	}

	feedResource.SpaceID = space.ID
	feedResource.FeedURI = feedURI
	feedResource.DownloadAttempts = downloadAttempts
	feedResource.DownloadRetryBackoffSeconds = downloadRetryBackoffSeconds
	feedResource.EnhancedMode = useExtendedAPI

	if len(feedUsername) > 0 {
		feedResource.Username = feedUsername
	}

	if len(feedPassword) > 0 {
		feedResource.Password = model.NewSensitiveValue(feedPassword)
	}

	feed, err := client.Feeds.Add(*feedResource)

	if err != nil {
		// TODO: handle error
	}
}
